### PR TITLE
Menu responsiveness

### PR DESCRIPTION
**Pull request by kickster97**
Side menu is now more responsive to smaller screens/browser windows. (<1000px)

Note: this PR uses font-awesome style sheets for the menu icons

Large window: 
image

Before:
image

Now with this PR: 
image

image

### CODE REVIEW

**Comment by carlhoerberg** [comment]
Very nice!

**Comment by viktorerlingsson** [comment]
Looks good! 

Maybe we could move the links to a var, to avoid having all links specified twice? Might be easy to miss to update both if we change anything. 

Something like this: 

```
var menuLinks= `
  <li><a id="menu-item" href="/">Overview</a></li>
  <li><a id="menu-item" href="/connections">Connections</a></li>
  <li><a id="menu-item" href="/channels">Channels</a></li>
  <li><a id="menu-item" href="/exchanges">Exchanges</a></li>
  <li><a id="menu-item" href="/queues">Queues</a></li>
  <li><a id="menu-item" href="/policies">Policies</a></li>
  <li><a id="menu-item" href="/operator-policies">Operator policies</a></li>
  <li><a id="menu-item" href="/shovels">Shovels</a></li>
  <li><a id="menu-item" href="/federation">Federation</a></li>
  <li><a id="menu-item" href="/vhosts">Virtual hosts</a></li>
  <li><a id="menu-item" href="/users">Users</a></li>
  <li><a id="menu-item" href="/nodes">Nodes</a></li>
  <li><a id="menu-item" href="/docs/" target="_blank">HTTP API</a></li>
`

document.getElementById('menu').innerHTML = `
<h1 id="menu-header">
  <a href="/"><img id="amq-logo" src="/img/logo-lavinmq-white.png"></a>
  <small id="version"></small>
  <small id="cluster_name"></small>
</h1>
<ul id="menu-content">` + menuLinks + `</ul>`

document.getElementById('small-menu').innerHTML = `
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
  <button id="menu-button" onclick=toggleSmallMenu()><i class="fa fa-lg fa-bars"></i></button>
<ul id="small-menu-content">` + menuLinks + `</ul>`

```

Haven't tested it much though, so don't know if it will cause any other issues :)

**Comment by kickster97** [comment]
> Looks good!
> 
> Maybe we could move the links to a var, to avoid having all links specified twice? Might be easy to miss to update both if we change anything.
> 
> Something like this:
> 
> ```
> var menuLinks= `
>   <li><a id="menu-item" href="/">Overview</a></li>
>   <li><a id="menu-item" href="/connections">Connections</a></li>
>   <li><a id="menu-item" href="/channels">Channels</a></li>
>   <li><a id="menu-item" href="/exchanges">Exchanges</a></li>
>   <li><a id="menu-item" href="/queues">Queues</a></li>
>   <li><a id="menu-item" href="/policies">Policies</a></li>
>   <li><a id="menu-item" href="/operator-policies">Operator policies</a></li>
>   <li><a id="menu-item" href="/shovels">Shovels</a></li>
>   <li><a id="menu-item" href="/federation">Federation</a></li>
>   <li><a id="menu-item" href="/vhosts">Virtual hosts</a></li>
>   <li><a id="menu-item" href="/users">Users</a></li>
>   <li><a id="menu-item" href="/nodes">Nodes</a></li>
>   <li><a id="menu-item" href="/docs/" target="_blank">HTTP API</a></li>
> `
> 
> document.getElementById('menu').innerHTML = `
> <h1 id="menu-header">
>   <a href="/"></a>
>   <small id="version"></small>
>   <small id="cluster_name"></small>
> </h1>
> <ul id="menu-content">` + menuLinks + `</ul>`
> 
> document.getElementById('small-menu').innerHTML = `
> <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
>   <button id="menu-button" onclick=toggleSmallMenu()><i class="fa fa-lg fa-bars"></i></button>
> <ul id="small-menu-content">` + menuLinks + `</ul>`
> ```
> 
> Haven't tested it much though, so don't know if it will cause any other issues :)

Good point, did some refactoring and fixed some smaller issues with the overview tab not getting highlighted. 
Now i think its ready!

**Review by viktorerlingsson — APPROVED** [review]

File: static/js/layout.js [hunk 1]
```diff
@@ -50,5 +46,56 @@ document.getElementById('user-menu').innerHTML = `
   </ul>
 `
 
-document.getElementById("userMenuVhost").onchange = (e) => Auth.selectVhost(e)
-document.getElementById("signoutLink").onclick = (e) => Auth.signOut(e)
+document.getElementById('small-menu').innerHTML = `
+<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
```
Comment (carlhoerberg) [inline static/js/layout.js:50]: No CDN! Everything has to be hosted locally.

Comment (carlhoerberg) [reply static/js/layout.js:50]: and stylesheets should not be injected dynamically

Comment (carlhoerberg) [reply static/js/layout.js:50]: is this only for the menu icon? they maybe just download that image and include, no need to inject a mega package of icons

Comment (kickster97) [reply static/js/layout.js:50]: Sure, will include images instead until we see that we need a range of icons then we can consider using a package.

**Pull request merged**